src/server/pps/server/api_server.go

```diff
@@ -742,6 +742,21 @@ func (a *apiServer) CreatePipeline(ctx context.Context, request *ppsclient.Creat
 		return nil, err
 	}
 
+	existingPipelineInfo, err := persistClient.GetPipelineInfo(ctx, request.Pipeline)
```

Comment:
What's the motivation for this block of code?  If the pipeline already exists, wouldn't we get an error when we call `CreatePipelineInfo` below?  This code can also race with two concurrent calls to `CreatePipelineInfo` (i.e. they both get `not found` and both move on to create the new pipeline; although they won't actually end up creating two pipelines since `CreatePipelineInfo` will prevent that).